extension/lib/magento/customer/getDiff.js

```diff
@@ -0,0 +1,37 @@
+const { differenceWith, isEqual } = require('lodash')
+/**
+ * @param {StepContext} context
+ * @param {Object} input
+ * @param {array} input.productIds - products to put
+ * @param {array} input.magentoProductIds - products currently on the magento fav list
+ * @returns {Promise<{productIds: Object}>}
+ */
+module.exports = async (context, input) => {
+  const { favoriteList, magentoProducts } = input
+  let favIds = []
+  let existIds = []
+
+  favoriteList.map(item => {
```

Comment:
yeah, as mentioned below, still using it as a foreach. This will be an exercise for u to refactor.